### ISSUE TITLE
Suggestion: replace `import *` with named imports

### ISSUE THREAD

**Issue by davidgilbertson** [fastai/fastbook#526]
I've just started the course and I'm finding that the use of `import *` makes learning more difficult than it should be. I don't know what functions are coming from where, and live in fear that I'll create a variable and accidentally overwrite something that's used elsewhere on the page.

In general, this practice is discouraged, [says the Python docs](https://docs.python.org/3/tutorial/modules.html#:~:text=Python%20programmers%20do%20not%20use%20this%20facility%20since%20it%20introduces%20an%20unknown%20set%20of%20names%20into%20the%20interpreter): "The practice of importing * from a module or package is frowned upon, since it often causes poorly readable code".

I'm just one opinion, but so far my impression of the fastai code snippets in the course is that they are indeed poorly readable, I would rather burn mental CPU cycles on understanding the concepts, not deciphering code.

BTW, I am aware of the `shift`+`tab` shortcut in Jupyter, and am aware I can type a function name and run a cell to see where it was defined, but that only half-solves half the problem.

**Comment by davidgilbertson** [comment]
Two seconds after logging this, I returned to the course, ran a cell, and got this:
![image](https://user-images.githubusercontent.com/4443482/183345720-9db53019-4e16-4220-a759-4e9e0d97eba5.png)

A slightly more rare example of why `import *` is poor form: this is quite difficult to troubleshoot, given that I have no idea where it's supposed to have come from, certainly more difficult than it should be.

**Comment by leszekhanusz** [comment]
Yes please, it is sometimes very confusing when you encounter some module, class or method which is not found above in the code.

**Comment by Abyszero** [comment]
Although I agree, this has been an issue from the very beginning and has been addressed during the MOOC in the past. It's not going to get fixed until there's a change of ownership. Best I can suggest is to rely on the docs instead of IDE code completion, and get to know the the library as well as you can.

**Comment by davidgilbertson** [comment]
It's weird, everything else about the course strikes me as very well thought out with clever ways to explain complex concepts. But then the code is littered with this sloppy and obfuscating `import *` habit that the rest of the community has worked out is a bad idea. Frustrating!

**Comment by programmer-ke** [comment]
As I software engineer getting into ML, I also noticed that notebooks tend to do * imports and initially found it odd (since in typical python coding in the software engineering context we tend to want to show where an import came from).

However, I've come to see it as some sort of convention in Data Science/ML notebooks that simplifies typing.

One thing that can help in knowing where something came from is checking the `__module__` attribute e.g. `vision_learner.__module__`.
